lib/nfc/protocols/st25tb/st25tb_poller_sync_api.c

```diff
@@ -0,0 +1,147 @@
+#include "nfc.h"
+#include "st25tb_poller_i.h"
+
+#include <nfc/nfc_poller.h>
+
+#include <furi.h>
+#include <stdint.h>
+
+#define ST25TB_POLLER_FLAG_COMMAND_COMPLETE (1UL << 0)
+
+typedef enum {
+    St25tbPollerCmdTypeDetectType,
+    St25tbPollerCmdTypeRead,
+    St25tbPollerCmdTypeReadBlock,
+    St25tbPollerCmdTypeWriteBlock,
+
+    St25tbPollerCmdTypeNum,
+} St25tbPollerCmdType;
+
+typedef union {
+    struct {
+        St25tbType* type;
+    } detect_type;
+    struct {
+        St25tbData* data;
+    } read;
+    struct {
+        uint8_t block_num;
+        uint32_t* block;
+    } read_block;
+    struct {
+        uint8_t block_num;
+        uint32_t block;
+    } write_block;
+} St25tbPollerCmdData;
```

Comment:
Please, avoid this kind of definitions. It would be much easier to read if you make an explicit type with typedef for each PollerCmdData and put them in union.

Making a new type is free from memory resources point of view, but makes your code much easier to read, maintain and prevents you from lot's of mistakes